src/olympia/blocklist/models.py

```diff
@@ -260,6 +259,10 @@ class BlocklistSubmission(ModelBase):
         help_text='The submission will not be published into blocks before this time.',
     )
     disable_addon = models.BooleanField(default=True)
+    block_type = models.IntegerField(
```

Comment:
I'm not a huge fan of this. In BlockVersion the field is a boolean called soft, and in BlockSubmissionList it is an integerField. Also, since BLOCK_TYPE_CHOICES is shared between these models (and will be used in MLBF eventually) I think we should extract this to a higher level, maybe module level on the models.py or in utils.

Maybe we could create a field for this to use in both models?

Also, I'm still not totally sure why we need the block_type on the submission level if I can submit multiple versions in a submission and there is a real use case of submitting different block types for versions of the same addon.

Why isn't a BlockSubmissionList creating Blocks -> BlockVersions ? I guess because we sometimes delay or await signoff etc etc. Still feels like an overly complicated approach opposed to a boolean field on the block "active" or something.. thinking out loud here but this doesn't feel right.

Reply:
We need `BlocklistSubmission` to handle multi-guids submissions, which is a frequent use-case for operations as when they find some malware, they often find more than one add-on connected to the same entity, and want to block them all in one go (or unblock if the decision needs to be reverted).

Since we need the delay/approval/disable add-on/notes/etc bits for those multi-guids block operations and also single-guid ones, for consistency and to avoid having the logic in several places, the admin was set up in such a way to make everything go through `BlocklistSubmission`. I'm simply following the logic here.

Reply:
Regarding integer vs boolean: calling it `block_type` here allowed me to directly expose it to the admin as a field admins can tweak and easily understand but I agree it can be confusing to us looking at the code...

If I move the `BLOCK_TYPE_CHOICES` to `src/olympia/constants/blocklist.py` with a comment explaining how it's shared, is that enough for you?

Reply:
yeah, pushing everything through `BlocklistSubmission` is necessary for a single submission that may be affecting multiple guid block(s), version(s), and other block details, but are gated by sign-off or delays.  It wouldn't _impossible_ to change everything around to do things differently, but it's a non-trivial amount of refactoring.

Reply:
Filed https://github.com/mozilla/addons/issues/15111

Reply:
4391116d2e3d6a73550798639c1429220c79cd3d introduces a different Choices... that shares the same name and re-uses the constants/values from `BlockVersion`. The true fix will be in https://github.com/mozilla/addons/issues/15111 which will make `BlockVersion` use `block_type` instead of the boolean (but IMHO out of scope here)